test/airport.spec.js

```diff
@@ -0,0 +1,331 @@
+airport = null;
+plane = null;
+expected = undefined;
+actual = undefined;
+result = undefined;
+
+console.log(`============================`);
+
+//Test 9 - try to take off a plane(id) which is not in the airport (landedPlanes)
+
+console.log(`============================`);
+console.log(`Test 9 - try to takeOffPlane Plane(id) that is not in landedPlanes`);
+
+// Arrange - this is setting up the variables required for the test
+airport = new Airport(2);
+plane1 = new Plane('Delta');
+expected = false;
+
+// Act - this is running the methods to be tested and collecting the actual results
+actual = airport.takeOffPlane(plane1);
+
+//Assert - testing the actual results against expected results
+result = assertEquals(expected, actual);
+console.log(`Test 9 - 'Plane is not present for take-off': ${result}`);
+
+// // Clean up - here we are clearing the variables ready for next test, to avoid contamination
+airport = null;
+plane = null;
+expected = undefined;
+actual = undefined;
+result = undefined;
+
+console.log(`============================`);
+
+//Test 10 - try to land a second instance of the same plane
+
+console.log(`============================`);
+console.log(`Test 10 - try to landPlane Plane(id) that is already in landedPlanes`);
+
+// Arrange - this is setting up the variables required for the test
+airport = new Airport(3); //set capacity higher than planes going in, so the capacity is not what blocks a re-landing plane
+plane1 = new Plane('Delta');
+expected = 1;
+
+// Act - this is running the methods to be tested and collecting the actual results
+airport.landPlane(plane1);
+airport.landPlane(plane1);
+actual = airport.getLandedPlaneCount(); //Boolean added to this function in the class, yes but in the end used getLandedPlaneCount!
+
+//Assert - testing the actual results against expected results
+result = assertEquals(expected, actual);
+console.log(`Test 10 - 'Plan cannot land if already landed': ${result}`);
+
+// // Clean up - here we are clearing the variables ready for next test, to avoid contamination
+airport = null;
+plane = null;
+expected = undefined;
+actual = undefined;
+result = undefined;
+
+console.log(`============================`);
+
+
+//Test 11 - test to check that a plane that has taken off, cannot take off again
+
+console.log(`============================`);
+console.log(`Test 11 - try to takeOffPlane Plane(id) that has already taken off`);
+
+// Arrange - this is setting up the variables required for the test
+airport = new Airport(1);
+plane1 = new Plane('Delta');
+expected = false;
+
+// Act - this is running the methods to be tested and collecting the actual results
+airport.landPlane(plane1); //land a plane at the airport first!
+//console.log(airport.isPlanePresent(plane1));
+airport.takeOffPlane(plane1);
+//console.log(airport.isPlanePresent(plane1));
+airport.takeOffPlane(plane1); 
+actual = airport.takeOffPlane(plane1); 
```

Comment:
Was thinking to use getPlaneCount, but this doesn't work if already gone to zero anyway.